### ISSUE TITLE
False errormsg:  "Not Delivered"

### ISSUE THREAD

**Issue by Jeepien** [signalapp/Signal-Android#3728]
Non-secure message to another cell phone kicks error  "Not Delivered"  but actually is delivered.  
[Tap for details] button yields message "Some issues need your attention" and under recipient says "Failed to send.". (But it did.)
[Resend] button does nothing.

**Comment by Jeepien** [comment]
Version used 2.22.2 on Verizontal Droid Maxx

**Comment by McLoo** [comment]
Please post a debug log.  right after that happens! 
 (settings  - advanced)

**Comment by johnwaldo** [comment]
This is happening to me as well. Either a 'not delivered' error message or a ... endless sending message. Debug posted.
https://gist.github.com/anonymous/e501a2a277ae8fd50cc5

**Comment by johnwaldo** [comment]
V: 2.23.3
Similar issues on wifi-compat and normal 3g. Receiving "Not Delivered" and an endless loop "..." as if its sending and waiting for confirmation of receipt.

https://gist.github.com/ac394fd21dc9cb2880ca

**Comment by moxie0** [comment]
@johnwaldo In your log, the system is not returning "success," it's returning RESULT_ERROR_NULL_PDU -- so we're displaying an error.  Are you running a rooted device or custom ROM?

**Comment by johnwaldo** [comment]
@moxie0 Hi Moxie. I am running stock android via samsung (default). 

Software Version: G900PVPU3BOF6
Hardware version: G900P.04
Android version: 5.0

The only thing that I can think of out of a standard "new in box/stock config" phone is that my device has been unlocked. Several months back I was going to install Cyanogenmod but lost interest when I read that certain key apps would not function on them. No additional steps have been taken down that road.

**Comment by Jeepien** [comment]
My phone is unmodified.

**Comment by riyapenn** [comment]
Possibly related to #4020 (where SmsSentJob shows message was delivered) and #3297

**Comment by riyapenn** [comment]
Also similar to https://github.com/WhisperSystems/TextSecure/issues/3014

**Comment by sedrubal** [comment]
:+1:

**Comment by heyimgay** [comment]
@sedrubal are you having the same problem? What is your configuration? Stock? Dual sim?

Do you have a debug log?

**Comment by sedrubal** [comment]
Yes, I have this problem in group chats (or at least in the only group chat I have :smile: )

I use `OmniRom 5.1.1` with one sim. `SMS and MMS` is `off`, `Privacy` is `off`, no linked devices.

My log captured right after receiving the "Not delivered" message: https://gist.github.com/anonymous/64a9d4c7b7d90418e675

**Comment by kaimast** [comment]
I only observe this with MMS messages (but every time I sent an MMS). Should I file a separate bug report for this?
This is on CyangenOS and T-Mobile US. SMS works fine though.

**Comment by 2-4601** [comment]
@kaimast There are quite many open tickets about issues with MMS on different providers. Try to have a look if someone has the same issue as yours and if not, file a separate bug report

**Comment by aidan-fitz** [comment]
@2-4601 can you add the bug label to this issue, à la #6002?

**Comment by jeremymasters** [comment]
One of the things I notice with this is that it's two specific users. Also, when I send a message, one check mark appears, then after a second, it says it's undelivered, but is actually delivered.

**Comment by rainerzufall** [comment]
A debug log after a false error message: https://gist.github.com/anonymous/4d1b044ca039aa3f4fcb03c235838bc3

What happened:
A group with 8 participants (including me and an unregistered number). Messages have been exchanged. no Problem. First one of my messages gets received 3 times by other group members. Later I get an error message, although my message was received 5 times on the phone of the other participants. When looking at my Desktop version my messages are not received at all. (desktop log: https://gist.github.com/anonymous/f728cc07bd01aebb5ab9ec55abe5a220)
I have not tinkered with any custom ROM or Backup thing.

Will https://github.com/WhisperSystems/Signal-Android/commit/7e51d61c7900e9d57c15eb6a6d595103bfdb7f79 solve this problem?
---
Device  : Fairphone FP2 (FP2)
Android : 5.1 (r4275.1_FP2_gms72_1.11.1, r4275.1_FP2_gms72_1.11.1)
Memory  : 89M (4.98% free, 128M max)
Memclass: 128
OS Host : SGSGP60SLV1
App     : Signal 3.27.1

**Comment by jeremymasters** [comment]
another data point: I sent an MMS to the person with whom I was having false SMS failures and lo and behold, no false failure.

TL;DR: SMS still fails to same person for me, but MMS seems to work fine.

**Comment by jeremymasters** [comment]
https://gist.github.com/6a3a24a059af21f75c596f1bd7012510

**Comment by riyapenn** [comment]
**Device:** Samsung Galaxy S7 
**Android version:** 7.0
**Signal version:** 4.9.9
**Provider:** Straight Talk via Verizon
**Log**: https://gist.github.com/c47d450dfb7daf096e5756ad15a9e4bb
SMS looks like it is not delivered, but actually is.  Changes to WiFi calling compatibility mode (Signal > Menu > Settings > SMS and MMS > and enable 'WiFi Calling' compatibility) or Delivery Reports (Signal > Settings > SMS and MMS > SMS delivery reports) made no difference.

**Comment by jeremymasters** [comment]
Galaxy s8
Android 7.0
Signal 4.14.10
Verizon to Verizon (iPhone) 
Issue is the same but the person receiving the message gets 10 messages every time!
https://gist.github.com/anonymous/c3520bc4459cf21fe95d3cc58d347f4f

**Comment by jeremymasters** [comment]
Galaxy s8 (different one than above)
Android 7.0
Signal 4.14.10
Verizon to verizon
https://gist.github.com/anonymous/348a1d2f061ad3bd30e5c66d9b681865
Sent message. Received no error. Received message. Sent another message, received error message

**Comment by automated-signal** [comment]
GitHub Issue Cleanup:
See #7598 for more information.
